feat(RELEASE-1442): allow merging tags array

jinqi7 opened the pull request:
This PR is to allow merging arrays recursively from collectors, Release,
ReleasePlan and ReleasePlanAdmission instead of overriding them.

Related PR: https://github.com/konflux-ci/release-service-utils/pull/524

## Describe your changes

## Relevant Jira

## Checklist before requesting a review
- [ ] I have marked as draft or added `do not merge` label if there's a dependency PR
  - If you want reviews on your draft PR, you can add reviewers or add the `release-service-maintainers` handle if you are unsure who to tag
- [ ] My commit message includes `Signed-off-by: My name <email>`
- [ ] I read CONTRIBUTING.MD and [commit formatting](CONTRIBUTING.md#commit-message-formatting-and-standards)
- [ ] I have run the README.md generator script in `.github/scripts/readme_generator.sh` and verified the results using `.github/scripts/check_readme.sh`

mmalina commented on tasks/managed/collect-data/collect-data.yaml:
```diff
@@ -231,6 +231,20 @@ spec:
         # Finally merge with ReleasePlanAdmission keys. ReleasePlanAdmission has higher priority
         merged_output=$(merge-json "$merged_output" "$release_plan_admission_result")
 
+        # Extract default tags from each source
```
The AC in Jira says:

- Merge-json script is updated to merge arrays instead of overriding the values.
- Collect-data task is updated with the latest release-service-utils image.

So it should be a generic solution, not just for this one use case.

jinqi7 replied:
OK. I didn't see the A/C at all since there was always some issue to load the whole ticket recently when I tried to access JIRA. I'll update the PR later.

jinqi7 replied:
https://github.com/konflux-ci/release-service-utils/pull/524

jinqi7 committed:
feat(RELEASE-1442): allow merging arrays recursively

Signed-off-by: Jing Qi <jinqi@redhat.com>

This PR is to allow merging arrays recursively from collectors, Release,
ReleasePlan and ReleasePlanAdmission instead of overriding them.

jinqi7 commented:
/retest

seanconroy2021 commented on tasks/managed/collect-data/tests/test-collect-data-tags.yaml:
```diff
@@ -0,0 +1,274 @@
+              set -eux
+
+              echo Test that the default tags in Release, ReleasePlan, ReleasePlanAdmission merged
+              v="$(jq -c '.mapping.defaults.tags' < "$(params.dataDir)/$(params.subdirectory)/data.json")"
```
Just a question what happens if you have latest in both default and in one of the CR is merged together or will it show twice.

jinqi7 replied:
No.  The tags are merged with duplicates removed
